index.js

```diff
@@ -266,11 +276,18 @@ class BlindPeer extends ReadyResource {
     }
     this.swarm.on('connection', this._onconnection.bind(this))
 
-    const announceProms = []
+    const startTime = Date.now()
```

Comment:
Can you extract it to a separate function? Then the logic can also be simplified, because it becomes linear

```
for await (const record of this.db.createAnnouncingCoresStream()) {
  if (this.closing) return
  await this._announceCore(record.key)
  if (this.closing) return
  await new Promise(resolve => setTimeout(resolve, this.announcingInterval)`
  if (this.closing) return
}
```


This also solves a lifecycle bug with the current logic: shutting the program down before all announces finished would leave pending timeouts still. It would shut down fully only after the the last scheduled timeout runs (and early-returns)